pkg/apis/core/validation/shoot.go

```diff
@@ -662,6 +713,23 @@ func ValidateKubernetesVersionUpdate(new, old string, skipMinorVersionAllowed bo
 	return allErrs
 }
 
+func validateNodeLocalDNSUpdate(newSpec, oldSpec *core.ShootSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if oldSpec.SystemComponents != nil && oldSpec.SystemComponents.NodeLocalDNS != nil &&
+		newSpec.SystemComponents != nil && newSpec.SystemComponents.NodeLocalDNS != nil &&
+		oldSpec.SystemComponents.NodeLocalDNS.Enabled != newSpec.SystemComponents.NodeLocalDNS.Enabled {
```

Comment:
What if `NodeLocalDNS.Enabled=true` in `old`, but `NodeLocalDNS` is now `nil` in `new`?